contracts/src/internal/HyperdriveLP.sol

```diff
@@ -103,6 +104,25 @@ abstract contract HyperdriveLP is
             revert IHyperdrive.InvalidEffectiveShareReserves();
         }
 
+        // Check to see whether or not the initial liquidity will result in
+        // invalid price discovery. If the spot price can't be brought to one,
+        // we fail to avoid dangerous pool states.
```

Comment:
Nit: “we revert” is clearer IMO since there are two meanings of “we fail to avoid”